test/testdrive/disk-feature-flag.td

```diff
@@ -51,7 +44,10 @@ r2 1 true
 $ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
 ALTER SYSTEM SET disk_cluster_replicas_default = true
 
-> CREATE CLUSTER c REPLICAS (r1 (SIZE '1', DISK), r2 (SIZE '1'))
+! CREATE CLUSTER c REPLICAS (r1 (SIZE '1', DISK), r2 (SIZE '1'))
+contains:DISK option not supported for non-legacy cluster sizes because disk is always enabled
+
+> CREATE CLUSTER c REPLICAS (r1 (SIZE '1'), r2 (SIZE '1'))
```

Comment:
Ah shoot, I'm nervous about losing all this test coverage. I see two options:

  1. Adjust the PR to set `is_cc: false` for the test-only sizes `1`, `2`, `1-1`, ... and so on. This preserves the current behavior of these test-only sizes. (I had switched them over to be `is_cc` since in self hosted all custom sizes are going to be marked as `is_cc`, but that broke all these tests—sorry!)
  2. Add a new size called `legacy` that is the one size that has `is_cc: false`.

Reply:
I'm also happier if we can keep the test diff smaller here, and keep the test coverage. I like option 1.

Reply:
Makes sense to me if you don't mind punching that in!

Reply:
Will do.